src/framework/mpas_bootstrapping.F

```diff
@@ -474,105 +480,173 @@ subroutine mpas_bootstrap_framework_phase2(domain) !{{{
       call mpas_log_write(' ')
       call mpas_log_write(' ')
 
-      ! Reading dimensions from streams
-      call mpas_log_write('Reading dimensions from input streams ...')
-      call mpas_stream_mgr_begin_iteration(domain % streamManager)
-      do while ( mpas_stream_mgr_get_next_stream(domain % streamManager, streamID = streamName, directionProperty = streamDirection, &
-                                                 activeProperty = streamActive) )
+      if (present(pio_file_desc)) then
+         call mpas_log_write('Reading dimensions from external PIO file handle ...')
+
+         ! Build stream dimension pool from the list of fields
+         call mpas_pool_create_pool(streamDimensions)
 
-         if ( streamActive .and. ( streamDirection == MPAS_STREAM_INPUT .or. streamDirection == MPAS_STREAM_INPUT_OUTPUT ) ) then
+         call mpas_pool_begin_iteration(domain % blocklist % allFields)
+         do while ( mpas_pool_get_next_member(domain % blocklist % allFields, poolItr) )
+            if ( poolItr % memberType == MPAS_POOL_FIELD ) then
+               call get_dimlist_for_field(domain % blocklist % allFields, poolItr % memberName, dimNames)
+               do i=1,size(dimNames)
+                  call mpas_pool_get_dimension(streamDimensions, dimNames(i), dimValue)
+                  if ( .not. associated(dimValue) ) then
+                     call mpas_pool_add_dimension(streamDimensions, dimNames(i), MPAS_MISSING_DIM)
+                  end if
+               end do
+            end if
+         end do
 
-            call mpas_stream_mgr_begin_iteration(domain % streamManager, streamID=streamName)
+         ioType = MPAS_IO_NETCDF  ! ioType is not actually used when an external PIO file_desc_t is provided to MPAS_io_open
+         inputHandle = MPAS_io_open('FILENAME_NOT_USED', MPAS_IO_READ, ioType, domain % ioContext, pio_file_desc = pio_file_desc, ierr = err_local)
 
-            ! Build stream dimension pool from the list of fields
-            call mpas_pool_create_pool(streamDimensions)
+         ! If to determine if file was opened or not.
+         if ( err_local == MPAS_IO_NOERR ) then
 
-            do while ( mpas_stream_mgr_get_next_field(domain % streamManager, streamName, fieldName, isActive=fieldActive) )
+            call mpas_log_write(' ')
 
-               if (fieldActive) then
-                  call get_dimlist_for_field(domain % blocklist % allFields, fieldName, dimNames)
+            ! Iterate over list of dimensions we determined we need from the above loop
```

Comment:
I think this comment would be a bit better if it explained why it was iterating: `! Iterate over the list of dimensions we determined we need from the above loop to read dimension values`.

What do you think?

Reply:
In my opinion, the comments in the loop body sufficiently explain what is happening when we iterate over the list of dimensions (which is more than just reading dimension values); for example, see the comments on [L.514](https://github.com/MPAS-Dev/MPAS-Model/blob/0b145f239cd54090bfba0c486017289ffaf6955d/src/framework/mpas_bootstrapping.F#L514) and [L.517](https://github.com/MPAS-Dev/MPAS-Model/blob/0b145f239cd54090bfba0c486017289ffaf6955d/src/framework/mpas_bootstrapping.F#L517).

Reply:
Sounds good.